src/controller/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+add_library(controller-ipc OBJECT ipc.cpp)
+
+target_link_libraries(controller-ipc
+    PUBLIC
+        nlohmann_json::nlohmann_json
+)
+
+find_package(Threads REQUIRED)
+find_package(Boost COMPONENTS filesystem REQUIRED)
+
+add_executable(controller 
+    controller
```

Comment:
My cmake (3.23.2) doesn't quite like the missing source file extensions here: Policy CMP0115 is not set: Source file extensions must be explicit.